core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -294,6 +294,9 @@ class TransactionStateManager(brokerId: Int,
           }
         }
 
+        replicaManager.forceCompleteDelayedProduce(
```

Comment:
Not sure if this is the best place to call force complete. For a regular produce request, if the leader changes when it's in the produce purgatory, it's also useful to force them to return immediately with a NotLeaderError. So, a better place to call this is probably from Partition.makeFollower().